File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.js
let input = require("fs").readFileSync("/dev/stdin").toString().trim().split(/\n/);

const haveCard = input[1].split(" ").map(Number);

let cardArr = new Map();
haveCard.forEach(n =>
  cardArr.has(n) ? cardArr.set(n, cardArr.get(n) + 1) : cardArr.set(n, 1)
);
const compareCard = input[3].split(" ").map(Number);
const res = compareCard.map(m => cardArr.get(m) || 0);
console.log(res.join(" "));

File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.py
from collections import defaultdict
import sys
input = sys.stdin.readline

N = int(input())
dic = defaultdict(int)
for num in input().strip().split():
    dic[num] += 1
    
M = int(input())
result = []
for num in input().strip().split():
    result.append(str(dic[num]))
print(' '.join(result))